Automattic/wp-calypso
# Fix all lint errors reported by `eslint-plugin-jest`

### Background

We have a few lint errors detected by `jest` plugin:

![image](https://user-images.githubusercontent.com/975703/94788435-d5824800-03d3-11eb-8120-19304b421fd3.png)


### Changes proposed in this Pull Request

* Fixes `jest/*` errors.
* Disables rules from "plugin:jest/recommended" for e2e tests as they use mocha

No more jest errors:
![image](https://user-images.githubusercontent.com/975703/94787721-f6966900-03d2-11eb-8dc5-93d2499e023c.png)

### Testing instructions

* Verify all tests pass (there may be unrelated lint errors)
* Run `./node_modules/.bin/eslint-nibble --ext .js,.jsx,.ts,.tsx .` (warning, it takes several minutes) and verify there are no violations for `jest/*` rules.
